mucommander-core/src/main/java/com/mucommander/ui/action/impl/ToggleTerminalAction.java

```diff
@@ -86,6 +86,13 @@ private void ensureIntegrationIsInitialized() {
         if (terminalIntegration == null && verticalSplit != null) {
             // doing it lazy, because in c-tor main frame might not be fully built (no vertical split pane yet)
             synchronized (LOCK) {
+                // TODO I'm lost - I see in debugger that finally this class is instantiated only once,
+                // but if I put breakpoint below I can see that mysteriously terminalIntegration
+                // is being set 2 times - first time, OK, good, but second time terminalIntegration
+                // (called via fireActivePanelChanged) is null again on the same instance of ToggleTerminalAction (!) - as if
+                // terminalIntegration was reset to null by reflection or clone, or sth?
+                // In Idea's debugger on the second occurrence there's a note that ToggleTerminalAction
+                // instance "is being initialized". ToggleTerminalAction has the same id!
```

Comment:
I see a similar behavior when debugging this from Eclipse but I don't see the constructor of TerminalIntegration being called twice - it looks like when placing a breakpoint in `terminalIntegration = new TerminalIntegration(mainFrame, verticalSplit);` then for some reason we get back to it when returning from the call to the aforementioned constructor.. @pskowronek when you put a breakpoint in the constructor of TerminalIntegration, do you see it called twice?

Reply:
> but I don't see the constructor of TerminalIntegration being called twice

Yup, that's why I was in desperation thinking that some cloned version appears or sth.

> when you put a breakpoint in the constructor of TerminalIntegration, do you see it called twice?

Yes, and that's why there some problems with Terminal (that there were 2 terminals initialized). That's why this helped in #1188: https://github.com/mucommander/mucommander/pull/1188/commits/7c94f2692abb542c10e6caed5fecdef51dd448ae#diff-0b49347bdb327817a5590941c5cdc9790e064980e06ee7801144a18a6bb644baR40

Reply:
I think the `volatile` variable might not behave as expected when it is called from the constructor, like in this case (since the object is not fully initialized yet). One way to work around this might be to have some static reference to a `TerminalIntegration` instance saved somewhere else

Reply:
> I think the `volatile` variable might not behave as expected when it is called from the constructor, like in this case (since the object is not fully initialized yet). One way to work around this might be to have some static reference to a `TerminalIntegration` instance saved somewhere else

nope, I changed:
```private volatile static TerminalIntegration terminalIntegration = null;```
to
```private TerminalIntegration terminalIntegration = null;```
and I see 2x invocations of TerminalIntegration.

I'm going to edit the comment just for state the behavior if volatile static is not used (`static` ensures that 
TerminalIntegration is not called twice).